### ISSUE TITLE
Upload to shared encrypted folder

### ISSUE THREAD

**Issue by MikeMatau** [dogbutcat/gclone#19]
What did I do:

Main Account with gdrive. In grdive there is a folder called /encrypted
The /encrypted (when mounted) contains the folders /Archiv and /transfer
Setup 100SA, put them in a groupd and shared the /encrypted folder with the group (inside the Main Account Drive UI)
I used rclone up to now, with the 10M bw limit. and the following config:

```
[gdrive]
type = drive
client_id = XXX
client_secret = XXX
scope = drive
token = {"access_token":XXX}

[gcrypt]
type = crypt
remote = gdrive:/encrypted
password = XXX
password2 = YYY
```

I then changed it to:

```
[gdrive]
type = drive
scope = drive
service_account_file = /home/XXX/accounts/YYY.json
service_account_file_path = /home/XXX/accounts/

[gcrypt]
type = crypt
remote = gdrive:/encrypted
password = XXX
password2 = YYY
```

The command for starting the move:

```
/usr/bin/gclone move /home/XXX/local/ gcrypt:/Archiv/ --delete-empty-src-dirs --log-file /home/XXX/rclone/logs/upload.log --check-first --drive-rolling-sa -L --drive-rolling-count=6 --drive-random-pick-sa --progress --timeout=0m30s --drive-shared-with-me
```
If I undersoot it correctly, the --drive-shared-with-me should make sure that the SA will use the encrypted folder that is shared with the group.
If I do:

```
/usr/bin/gclone lsd gcrypt: --drive-rolling-sa -L --drive-random-pick-sa --progress --timeout=0m30s --drive-shared-with-me
```
I get the list of folders I want to see:

/
/Archiv
/Archiv/FolderA
/Archiv/FolderB ...
/transfer

But somehow..... The move command transered quite a lot of files (encrypted) to the SA gdrive, and not the shared folder...
Now all 100 SA's gdrive are filled with random files, that should be in the shared /encrypted folder...

Now I'm wondering about 3 things:

1. How do I move all the files from the 100 SA's to the shared folder...?
2. Will it work when I just move everything from the SA to the shared? Or do I have to mount each SA and decrypt and move to the proper folder? If yes... How?
3. How come, some of the transfered folders where moved to the shared folder and are working fine on another server mounting the Main Account gdrive and decrypting?

Is it possible that all uploads to a SA will be added to their 15GB storage, even if it is in a shared folder? Or did something else happen?

I'm realy confused....

**Comment by dogbutcat** [comment]
Thank you for asking, I've made a lot of time to understanding and research for your situation, now I have several questions if you could answer:

1. I want to confirm that it work fine before you changed to service account logic?
2. Is your [gdrive] a personal drive? not team drive I mean.
3. If you want to expand your google drive space using SA?

if all above is YES. In my opion, your result was right. 

1. SA is suitable for team drive. Let's first look what is [SA from google](https://cloud.google.com/iam/docs/service-account-overview), `A service account is a special kind of account typically used by an application or compute workload`. So it shouldn't be access from user scope, at least I don't know how to access from google web.
1. Let's talk about `--drive-shared-with-me`, from comment in rclone 
https://github.com/dogbutcat/gclone/blob/9ab05b103b8c71acd5d896a06b7393f2919d06a9/backend/drive/drive.go#L922-L925
the content list is the same with that displayed in google drive web under `Shared with me` tab. Back to opt 1, SA can't access from google web, also can't access `Shared with me`, at least I haven't found how to.

last, your question, from my view, you are decide to store archive data in "YOUR" drive. I have a idea, but you need at least one "TEAM" drive to bridge.

1. add SA group to "TEAM" drive and create folder `/encrypted`.
2. create shortcuts from "YOUR" drive to "TEAM" drive's `encrypted` folder, like `/transfer`、`/Archiv`、 etc...
3. replace your `[gdrive]` with below
```
[gdrive]
type = drive
scope = drive
service_account_file = /home/XXX/accounts/YYY.json
service_account_file_path = /home/XXX/accounts/
root_folder_id =
team_drive = [TEAM_DRIVE_ID]
```
now you can run your command to see if work.

I don't know if SA has 15G storage limit as it acts for drive api resource read/write. Please remember SA is not a "USER" role, best experience with team drive.

**Comment by MikeMatau** [comment]
Thank you for taking the time to read and understand my issue....
First of all:

1. I want to confirm that it work fine before you changed to service account logic?:
Yes, before it was working as intended (with the 10mbit limit to prevent reaching 750GB/Day)

2. Is your [gdrive] a personal drive? not team drive I mean.
Yes, it is just a personal drive

3. If you want to expand your google drive space using SA?
No, the goal was just to be able to upload more than 750GB because I sometimes have spikes where many GB's need to be transfered in a short period of time

So, I guess the exact problem was: I thought I would be able to upload into my shared Folder, without using up the SA quota.
Instead, my uploads to the SA where put in the shared folder, but used up the quota and are also present in each SA drive.
Basically I have now 1.5TB spread across 100 drives who are all sharing these files with my main drive..........

So now, I need to setup a teamdrive (how?). Do I also need to transfer all my files from my personal drive to the teamdrive if I want to use that in future so I'm able to use SA's?
Because I think, If I copy everything from my personal drive to my gdrive (including the files that are "shared" from the SA's) I can delete all files from the SA's afterwards (somehow).

The only issue I see: we are talking about 270TB ......

**Comment by dogbutcat** [comment]
close for in-active, for further questions could open again.

Thanks.
